pkg/clustermanager/provision_node.go

```diff
@@ -139,7 +152,7 @@ func (provisioner *NodeProvisioner) prepareDocker() error {
 	// docker-ce
 	aptPreferencesDocker := `
 Package: docker-ce
-Pin: version 17.03.*
+Pin: version 18.03.*
```

Comment:
This shouldn't be changed. This was used to pin the Docker CE release to prevent automatic upgrades to newer versions which aren't compatible with K8S.

See also: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.10.md#external-dependencies

Reply:
I'm aware of that, for now 18.03 is the only version in the stable repository from docker ce afaik. Perhaps we should use the docker.io package, I think it's on 17.03

Reply:
- The Ubuntu/bionic provided [docker.io package](https://packages.ubuntu.com/bionic/docker.io) is at 17.12.1, no older versions available :red_circle: 
- The Docker CE package is at [18.03](https://download.docker.com/linux/ubuntu/dists/bionic/stable/binary-amd64/Packages) :red_circle:

None of those versions is compatible.

Another option would be to switch to [cri-o](http://cri-o.io/) and thereby finally getting rid of Docker…

Reply:
@JohnnyQQQQ docker-ce 17.03 should be available in the repo cause this bellow works. 
```
export docker_version=17.03.2
apt-get install -y -q docker-ce=$(apt-cache madison docker-ce | grep ${docker_version}
OR
install -y -q --setopt=obsoletes=0 docker-ce-${docker_version}.ce
```

Reply:
@qubusp I still only get versions > 17

Reply:
I have subscribed to https://github.com/kubernetes/kubernetes/issues/65089 and will wait for any updates

Reply:
K8s 1.10 supports Docker 17.3.x if I'm not wrong: https://kubernetes.io/docs/imported/release/notes/
`https://kubernetes.io/docs/imported/release/notes/`

Reply:
You can install docker 17.03 on ubuntu 18.04 using snap:

``` bash
snap install docker --channel=17.03/stable
```

Reply:
no, it's not a good idea to use snap. I can't run privileged containers and it's not running in systemd.

About cri-o, there is also no support for 18.04, yet. So this won't help us in this problem anyways. 

I fear we need to get up a stable runtime first, in a supported way. That looks somehow really complicated at the moment :disappointed:

Reply:
referring to my last comment, we need to pin 18.06 

and use the config I mentioned lower